### PR TITLE
Improve Doxygen comments in n_request.c.

### PR DESCRIPTION
**Pull request by haydenroche5**
Also, make them all conform to the same style.

This is a step toward generating the API docs for note-c. I'll tackle the other source files next, and eventually will come up with a scheme for generating and publishing the docs somewhere.

### CODE REVIEW

File: n_request.c [hunk 1]
```diff
@@ -587,20 +590,19 @@ void NoteErrorClean(char *begin)
...
+ * @param p The hex string to convert.
+ * @param len The length of the hex string.
+ *
+ * @returns The converted number.
+ */
+uint64_t n_atoh(char *p, int len)
 {
     uint64_t n = 0;
-    char *ep = p+maxlen;
+    char *ep = p+len;
```
Comment (haydenroche5) [inline n_request.c:605]: Not sure why this was called maxlen before. It's just the length of the passed in buffer, as far as I can tell. Max seems superfluous.

File: n_request.c [hunk 2]
```diff
@@ -750,4 +751,5 @@ NOTE_C_STATIC bool crcError(char *json, uint16_t shouldBeSeqno)
     uint32_t shouldBeCrc32 = crc32(json, fieldOffset);
     return (shouldBeSeqno != actualSeqno || shouldBeCrc32 != actualCrc32);
 }
-#endif
+
+#endif // !NOTE_LOWMEM
```
Comment (haydenroche5) [inline n_request.c:755]: There was an `#ifdef` guard on each of the CRC functions. I simplified it by just using one for all of them as a group.

**Comment by zfields** [comment]
Can you resolve the conflicts now that we've reworked a bunch of stuff?

File: n_request.c (hunk 1 above)
Comment (zfields) [reply n_request.c:605]: I would guess it has to do with lines numbers 612, 613, and 614.

**Comment by haydenroche5** [comment]
> Can you resolve the conflicts now that we've reworked a bunch of stuff?

Yep. Should be good to go now.

File: n_request.c [hunk 3]
```diff
@@ -648,20 +644,19 @@ void NoteErrorClean(char *begin)
...
-/**************************************************************************/
-uint64_t n_atoh(char *p, int maxlen)
+ @brief Convert a hex string to a 64-bit unsigned integer.
+
+ @param p The hex string to convert.
+ @param maxLen The length of the hex string.
+
+ @returns The converted number.
+ */
+uint64_t n_atoh(char *p, int maxLen)
```
Comment (zfields) [inline n_request.c:656]: Did you also update the declaration?

**Review by zfields — APPROVED** [review]
Approved with comments.

File: n_request.c [hunk 4]
```diff
@@ -570,63 +573,56 @@ J *noteTransactionShouldLock(J *req, bool lockNotecard)
...
-  @param   errstr
-  The error string.
-  @param   errtype
-  The error type string.
-  @returns boolean. `true` if the string contains the error provided, `false`
-  if not.
-*/
-/**************************************************************************/
+ @brief Check to see if a Notecard error is present in a JSON string.
+
```
[the remark this answers is not in the document]
Comment (zfields) [reply n_request.c:632]: I was hoping you would fashion that into the long description, similar to `NoteErrorClean()` below.

FWIW, I normally throw @notes at the bottom, not that it matters.

**Review by zfields — APPROVED** [review]

**Pull request merged**